`TrialOrIntroPriceEligibilityChecker`: fixed integration test to ensure receipt is always loaded

NachoSoto opened the pull request:
The reason the receipt data is required, is because if it's `nil`, it ends up becoming empty because of `data ?? Data()`, but `GetIntroEligibilityOperation` never actually runs because that's empty.

To make sure the test doesn't fail, we begin by ensuring the receipt data is loaded first.

Also documented the reason for `ReceiptRefreshPolicy.never`.

taquitos reviewed (approved):
🤯 🐐 

![image](https://user-images.githubusercontent.com/1304821/174907143-a22ab6d3-a5b2-4e38-bdcc-558bf2718bb3.png)

NachoSoto commented on Sources/Purchasing/TrialOrIntroPriceEligibilityChecker.swift:
```diff
@@ -82,7 +82,7 @@ class TrialOrIntroPriceEligibilityChecker {
     func sk1CheckEligibility(_ productIdentifiers: [String],
                              completion: @escaping ReceiveIntroEligibilityBlock) {
-        receiptFetcher.receiptData(refreshPolicy: .never) { data in
+        receiptFetcher.receiptData(refreshPolicy: .onlyIfEmpty) { data in
```
cc @aboedo any reason why this should be `.never`?

aboedo replied:
yeah, sadly. 
refreshing here will likely prompt the user for their credentials, and intro eligibility should be triggered programmatically. 

In theory, this should only be a problem in sandbox, but I have a vague recollection of having moved to `never` after this caused a lot of customer confusion and headaches

aboedo replied:
that makes total sense. 

Also, would you mind adding what I explained as a comment? Your question about why this should be `.never` is very valid and there's no reason maintainers should know the explanation before seeing this code, I should have added a comment myself when adding the `.never` 🤦

NachoSoto commented:
@aboedo I've updated this:
- Change implementation to only sandbox
- Added docs
- Added a unit test to cover this

Because Integration Tests aren't detecting "sandbox" I still need those tests separately.

NachoSoto commented on Sources/Purchasing/TrialOrIntroPriceEligibilityChecker.swift:
```diff
@@ -82,7 +82,15 @@ class TrialOrIntroPriceEligibilityChecker {
+        // We don't want to always refresh receipts in production because it will
+        // likely prompt the user for their credentials, and intro eligibility should be triggered programmatically.
+        let refreshPolicy: ReceiptRefreshPolicy = self.systemInfo.isSandbox
+            ? .onlyIfEmpty
```
Just rebased. Integration tests will use this now :)

NachoSoto commented:
All tests passing 🎉
